bare_metal_billing/s3_bucket.py

```diff
@@ -0,0 +1,33 @@
+import os
+import logging
+import functools
+
+import boto3
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+@functools.lru_cache
+def get_lease_bucket():
+    try:
+        s3_resource = boto3.resource(
+            service_name="s3",
+            endpoint_url=os.environ.get(
+                "S3_ENDPOINT", "https://s3.us-east-005.backblazeb2.com"
+            ),
+            aws_access_key_id=os.environ["S3_KEY_ID"],
+            aws_secret_access_key=os.environ["S3_APP_KEY"],
+        )
+    except KeyError:
+        logger.error(
+            "Error: Please set the environment variables S3_KEY_ID and S3_APP_KEY"
+        )
+    return s3_resource.Bucket(os.environ.get("S3_BUCKET_NAME", "esi-leases"))
```

Comment:
I've learned this the hard way, but I wouldn't assign a default bucket name. This is so when it's actually deployed to run automatically it would be nice to see to be explicit about what bucket we are getting the lease info from.